### PR TITLE
DHFPROD-4670: Add hub-central-flow-writer role

### PR DESCRIPTION
**Pull request by ryanjdew**
### Description
Also:
-  adjusted the privileges to data-hub-common to be able to call mlHubversion extension
- fixed an issue where flow-operator was needed in addition to hub-central-step-runner on ML 9.0-12

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/FlowMvcTest.java [hunk 1]
```diff
@@ -25,12 +25,7 @@
     @Test
     void runIngestionStep() throws Exception {
         installReferenceModelProject();
-        if(isVersionCompatibleWith520Roles()) {
-            loginAsTestUserWithRoles("hub-central-step-runner");
-        }
-        else {
-            loginAsTestUserWithRoles("hub-central-step-runner", "flow-operator-role");
-        }
+        loginAsTestUserWithRoles("hub-central-step-runner");
```
Comment (bsrikan) [inline marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/FlowMvcTest.java:28]: Is this change intentional. Srinath fixed it so test will pass in 9.0-12.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.test.tsx [hunk 2]
```diff
@@ -126,4 +127,74 @@ describe("Entity Tiles component", () => {
...
+    // test adding to existing flow
+    expect(getByText('Add step to an existing flow')).toBeInTheDocument();
+    fireEvent.click(getByText('Select Flow'));
+    fireEvent.click(getByText(data.flows.data[0].name));
+    fireEvent.click(getByText('Yes'));
+    expect(mockAddStepToFlow).toBeCalledTimes(1);
+
+    // adding to new flow
+    fireEvent.mouseOver(getByText(mapping[0].name));
+    expect(getByText('Add step to a new flow')).toBeInTheDocument();
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.test.tsx:165]: Nice. Can we also add a check for "Add step to an existing flow"

File: marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/FlowMvcTest.java (hunk 1 above)
Comment (ryanjdew) [reply marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/FlowMvcTest.java:28]: Yes. I adjusted hub-central-step-runner to work with both 9.0-12 and 10.0-*

File: marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx [hunk 3]
```diff
@@ -1,10 +1,85 @@
...
+    expect(getByText('Add step to an existing flow')).toBeInTheDocument();
+    fireEvent.click(getByText('Select Flow'));
+    fireEvent.click(getByText(data.flows.data[0].name));
+    fireEvent.click(getByText('Yes'));
+    expect(mockAddStepToFlow).toBeCalledTimes(1);
+
+    // adding to new flow
+    fireEvent.mouseOver(getByText(data.loads.data[0].name));
+    expect(getByText('Add step to a new flow')).toBeInTheDocument();
+    // TODO calling addStepToNew not implemented yet
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx:54]: Same as mapping, can we include "Add to existing flow" assertion.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.test.tsx (hunk 2 above)
Comment (ryanjdew) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.test.tsx:165]: "Add step to an existing flow" is above. It doesn't appear that "Add step to a new flow" is fully implemented yet.

File: marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/FlowMvcTest.java (hunk 1 above)
Comment (bsrikan) [reply marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/FlowMvcTest.java:28]: Thanks Ryan. I deleted the other comment after seeing the existing flow was asserted and it seems to have deleted your comment as well. Sorry about that.

File: marklogic-data-hub-central/ui/src/pages/Run.tsx [hunk 4]
```diff
@@ -34,7 +34,7 @@ const Run: React.FC = () => {
     const authorityService = useContext(AuthoritiesContext);
     const canReadFlow = authorityService.canReadStep();
     const canWriteFlow = authorityService.canWriteStep();
-    const hasOperatorRole = authorityService.hasOperatorRole();
+    const hasOperatorRole = authorityService.canRunStep();
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/pages/Run.tsx:37]: Currently we have tests for Run tile in Run.test.tsx, but they are run as developer and operator. Should we run them as flow-writer and step runner instead, so we would have verified a flow can be run with a lesser and more appropriate authority?

File: marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx [hunk 5]
```diff
@@ -106,8 +106,10 @@ describe('TilesView component', () => {
 
     });
 
-    test('Verify Run tile displays from toolbar', async () => {
-        const {getByLabelText, getByText, queryByText, getByTestId} = await render(<AuthoritiesContext.Provider value={ mockDevRolesService}><TilesView/></AuthoritiesContext.Provider>);
+    test('Verify Run tile displays from toolbar and can create/edit/delete with writeStep authority', async () => {
+        const authorityService = new AuthoritiesService();
+        authorityService.setAuthorities(['readStep','writeStep','runStep']);
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx:111]: Nice. I should probably comment after am done reviewing everything top down.
With this I think we can keep the Run tests as is and you can ignore my comment earlier regarding Run.test.

File: marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx [hunk 6]
```diff
@@ -136,7 +138,7 @@ describe('TilesView component', () => {
         expect(getByTestId('runStep-1')).toBeInTheDocument();
     });
 
-    test('Verify run tile cannot edit with only readStep authority', async () => {
+    test('Verify run tile cannot edit or run with only readStep authority', async () => {
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx:141]: Can we add a custom step to run.config and verify it shows up in Run tile for these authorities?

File: marklogic-data-hub-central/ui/src/pages/Run.tsx (hunk 4 above)
Comment (ryanjdew) [reply marklogic-data-hub-central/ui/src/pages/Run.tsx:37]: wouldn't hurt. I'll adjust that.

File: marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx (hunk 6 above)
Comment (ryanjdew) [reply marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx:141]: sure

**Comment by bsrikan** [comment]
> Should there be tests where user with this role can create and delete flows from "Run" page ?

Yes I believe we should include that. Its listed in test plan as well "View and CRUD Run tile"

File: marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx (hunk 6 above)
Comment (ryanjdew) [reply marklogic-data-hub-central/ui/src/pages/TilesView.test.tsx]: added a custom step and a mapping step

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/FlowController.java [hunk 7]
```diff
@@ -54,32 +54,37 @@
     @RequestMapping(method = RequestMethod.POST)
     @ResponseBody
     @ApiOperation(value = "Create a flow", response = FlowInfo.class)
+    @Secured("ROLE_writeStep")
```
[the remark this answers is not in the document]
Comment (ryanjdew) [reply marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/FlowController.java]: Refactored these to be readFlow/writeFlow

**Comment by srinathgit** [comment]
PR requires a rebase
